src/useKey/useKey.ts

```diff
@@ -0,0 +1,41 @@
+import { DependencyList, RefObject, useMemo } from 'react';
+import { useEventListener } from '..';
+import { noop, isBrowser } from '../util/const';
+
+export type IKeyPredicate = (event: KeyboardEvent) => boolean;
+export type IKeyFilter = null | undefined | string | IKeyPredicate;
+export type IHandler = (event: KeyboardEvent) => void;
+
+export type IUseKeyOptions<T extends EventTarget> = {
+  event?: 'keydown' | 'keypress' | 'keyup';
+  target?: RefObject<T> | T | null;
+  options?: boolean | AddEventListenerOptions;
+};
+
+const createKeyPredicate = (keyFilter: IKeyFilter): IKeyPredicate => {
+  if (typeof keyFilter === 'function') return keyFilter;
+  if (typeof keyFilter === 'string') return (event: KeyboardEvent) => event.key === keyFilter;
+  return keyFilter ? () => true : () => false;
```

Comment:
Move these to `src/util/misc.ts` name it like `yieldTrue` and `yieldFalse`.

What you do here is create new function for each hook that is not optimal in terms of memory consumtion, it is nearly memory leackage.